Tasks/DownloadBuildArtifact/main.ts

```diff
@@ -0,0 +1,48 @@
+import {BuildArtifact, ArtifactResource} from 'vso-node-api/interfaces/BuildInterfaces';
+import {WebApi, getBearerHandler} from 'vso-node-api/WebApi';
+import * as tl from 'vsts-task-lib/task';
+
+import {ArtifactProvider} from './ArtifactProvider';
+import {FileContainerProvider} from './FileContainer';
+
+async function main(): Promise<void> {
+	let projectId = tl.getVariable('System.TeamProjectId');
+	let buildId = parseInt(tl.getInput("buildId"));
+	let artifactName = tl.getInput("artifactName");
+	let downloadPath = tl.getPathInput("downloadPath");
+
+	let accessToken = getAuthToken();
+	let credentialHandler = getBearerHandler(accessToken);
```

Comment:
Can you run this task locally interactively to test / debug?

In the step by step I encourage that:
https://github.com/Microsoft/vsts-task-lib/blob/master/node/docs/stepbystep.md

In order to pull it off, I have to support a pat (interactive) or bearer (run in a job).

See here.  Let me know if you have any better ideas.

https://gist.github.com/bryanmacfarlane/154f14dd8cb11a71ef04b0c836e5be6e#file-vsts_sample-ts